src/adapter/graphql/src/queries/accounts/account_flow_runs.rs

```diff
@@ -105,12 +96,44 @@ impl AccountFlowRuns {
             total_count,
         ))
     }
+
+    async fn list_datasets_with_flow(&self, ctx: &Context<'_>) -> Result<DatasetConnection> {
+        let flow_service = from_catalog::<dyn fs::FlowService>(ctx).unwrap();
+
+        let datasets_with_flows: Vec<_> = flow_service
+            .list_all_datasets_with_flow_by_account(&self.account.id)
+            .await
+            .int_err()?
+            .matched_stream
+            .try_collect()
+            .await?;
+
+        let matched_datasets: Vec<_> =
+            futures::future::try_join_all(datasets_with_flows.iter().map(|dataset_id| {
+                let dataset_ref = dataset_id.as_local_ref();
+                async move { Dataset::from_ref(ctx, &dataset_ref).await }
+            }))
+            .await?;
```

Comment:
I think this might be too expensive. ID->HDL resolution is relatively fast with S3, as it matches the layout on the disk, however it's expensive with local-FS, as each of these is scanning datasets directory for every dataset.

Think of a batch opreation that would take a collection of `dataset_id` and return you a collection of handles, ensuring a single scan of the underlying storage system.